### PR TITLE
Blast doors & shutters fixes & cleanup: toolspeed, sounds, player feedback

### PR DESCRIPTION
**Pull request by Sealed101**
## About The Pull Request
Blast doors & shutters were using do_after for their deconstruction steps without taking toolspeed into the account. This PR replaces this by use_tool, so better tools actually do the steps faster. It also brings back the sounds for each step.
Screwdrivering the panel open/closed no longer produces two messages.
Moved two defines to their rightful place.
Each step now also provides feedback to the player upon completion. It used to just give you the _You start ..._ but no _You finish ..._ Wack.

## Why It's Good For The Game
Better tools should actually do their job better. Also, sounds and more feedback good.

## Changelog
:cl:
fix: Better tools now deconstruct blast doors and shutters quicker like they actually should. Each deconstruction step now provides a message upon completion as well as uses the tool's sounds!
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/poddoor.dm [hunk 1]
```diff
@@ -43,30 +36,35 @@
 
 		else if(W.tool_behaviour == TOOL_CROWBAR && deconstruction == BLASTDOOR_FINISHED)
 			to_chat(user, span_notice("You start to remove the airlock electronics."))
-			if(do_after(user, 10 SECONDS, target = src))
+			if(W.use_tool(src, user, 100, volume=50))
 				new /obj/item/electronics/airlock(loc)
 				id = null
 				deconstruction = BLASTDOOR_NEEDS_ELECTRONICS
+				to_chat(user, span_notice("You remove the airlock electronics."))
```
Comment (Mothblocks) [inline code/game/machinery/doors/poddoor.dm:43]:
```suggestion
				balloon_alert(user, "removed airlock electronics")
```

File: code/game/machinery/doors/poddoor.dm [hunk 2]
```diff
@@ -43,30 +36,35 @@
...
 
 		else if(W.tool_behaviour == TOOL_WIRECUTTER && deconstruction == BLASTDOOR_NEEDS_ELECTRONICS)
 			to_chat(user, span_notice("You start to remove the internal cables."))
-			if(do_after(user, 10 SECONDS, target = src))
+			if(W.use_tool(src, user, 100, volume=50))
 				var/datum/crafting_recipe/recipe = locate(recipe_type) in GLOB.crafting_recipes
 				var/amount = recipe.reqs[/obj/item/stack/cable_coil]
 				new /obj/item/stack/cable_coil(loc, amount)
 				deconstruction = BLASTDOOR_NEEDS_WIRES
+				to_chat(user, span_notice("You remove the internal cables."))
```
Comment (Mothblocks) [inline code/game/machinery/doors/poddoor.dm:53]:
```suggestion
				balloon_alert(user, span_notice("removed internal cables"))
```

File: code/game/machinery/doors/poddoor.dm [hunk 3]
```diff
@@ -43,30 +36,35 @@
...
 
 			to_chat(user, span_notice("You start tearing apart the [src]."))
-			playsound(src.loc, 'sound/items/welder.ogg', 50, 1)
-			if(do_after(user, 15 SECONDS, target = src))
+			if(W.use_tool(src, user, 150, volume=50))
 				var/datum/crafting_recipe/recipe = locate(recipe_type) in GLOB.crafting_recipes
 				var/amount = recipe.reqs[/obj/item/stack/sheet/plasteel]
 				new /obj/item/stack/sheet/plasteel(loc, amount)
 				qdel(src)
+				to_chat(user, span_notice("You tear the [src] apart."))
```
Comment (Mothblocks) [inline code/game/machinery/doors/poddoor.dm:66]:
```suggestion
				balloon_alert(user, "torn apart")
				qdel(src)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Balloon alerts should be favored over to_chat where archival is not necessary.

https://hackmd.io/@tgstation/SJIle5wYd

**Comment by Sealed101** [comment]
confused_seal_doing_questionable_life_choices.mp4

https://user-images.githubusercontent.com/75863639/131928783-3b41270f-d200-4646-9a4b-6bff63ac60d7.mp4

**Comment by Mothblocks** [comment]
hahaha wtf

**Comment by TemporalOroboros** [comment]
> confused_seal_doing_questionable_life_choices.mp4
> 
>  balloonacy.mp4

I am guessing that the balloon alert looks for the location of the thing doing the balloon alert and then spawns the pop-up on that. It looks like it doesn't account for balloon alerts on turfs and is spawning the pop-up on the area.

**Comment by TemporalOroboros** [comment]
https://github.com/tgstation/tgstation/blob/feb0661e2281a1b15f66b8554011186cfda77a1b/code/__HELPERS/unsorted.dm#L331-L340

Actually, it looks like this goes all the way to `get_atom_on_turf` not accounting for being called on a turf. In theory the same thing should happen with runechat if someone calls `Say(...)` on a turf unless there are additional checks there.

**Comment by TemporalOroboros** [comment]
`Say(...)` is an `/atom/movable` level proc. Disregard.

**Comment by Mothblocks** [comment]
temporal is right, i knew it was somehow getting the area

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**